### PR TITLE
Feature/issue 1134 add review modal

### PR DESCRIPTION
**Pull request by brandon-PG**

### CODE REVIEW

File: src/stores/__tests__/user_profiles.test.ts [hunk 1]
```diff
@@ -208,4 +208,19 @@ describe('user profile reducer', () => {
             expect(newStore.bookmarkBanner).toBe(BookmarkBanner.RegionHasNotBeenChanged);
         });
     });
+
+    describe('the timesAppStarted number', () => {
+
+        test('is increased by application started action', () => {
+            const oldStore = new UserProfileBuilder().build();
+            const newStore = reducer(oldStore, applicationStarted());
+            expect(newStore.timesAppStarted).toBe(oldStore.timesAppStarted + 1);
```
Comment (rasmus-storjohann-PG) [inline src/stores/__tests__/user_profiles.test.ts:217]: JS doesn't have an integer type, so all numbers are floating point, which are tricky to compare for equality

File: src/selectors/user_profile/select_times_app_started.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { Store } from '../../stores';
+
+export const selectTimesAppStarted = (appStore: Store): number => {
+    return appStore.userProfile.timesAppStarted;
+};
```
Comment (rasmus-storjohann-PG) [inline src/selectors/user_profile/select_times_app_started.ts:5]: This is not our coding style for one line functions

File: src/stores/__tests__/user_profiles.test.ts [hunk 3]
```diff
@@ -222,5 +222,15 @@ describe('user profile reducer', () => {
...
+
+        it('is loaded from persisted data', () => {
+            const oldStore = new UserProfileBuilder().build();
+            const persistedDataBuilder = new PersistedDataBuilder().
+                withTimesAppStarted(aNumber()).
+                build();
+            const presistanceAction = DataPersistence.loadSuccess(persistedDataBuilder);
+            const newStore = reducer(oldStore, presistanceAction);
+            expect(newStore.timesAppStarted).toBe(persistedDataBuilder.timesAppStarted);
+        });
```
Comment (rasmus-storjohann-PG) [inline src/stores/__tests__/user_profiles.test.ts:234]: Nice

File: src/components/main/main_connected_component.ts [hunk 4]
```diff
@@ -55,6 +56,7 @@ const mapStateToProps = (store: Store, ownProps: RouterProps): Props => ({
     isCreatorModalVisible: selectIsCreatorModalVisible(store),
     feedbackScreen: selectFeedbackScreen(store),
     theme: selectUxTheme(store),
+    showReviewModal: selectTimesAppStarted(store) === 2,
```
Comment (rasmus-storjohann-PG) [inline src/components/main/main_connected_component.ts:59]: I don't want that kind of low level detail in a connected component file, it should be high level "connecting" only. Make a selector that computes whether to show the modal or not. This selector can replace the existing `selectTimesAppStarted()`

File: src/stores/__tests__/user_profiles.test.ts [hunk 5]
```diff
@@ -217,10 +217,10 @@ describe('user profile reducer', () => {
             expect(newStore.timesAppStarted).toBe(oldStore.timesAppStarted + 1);
         });
 
-        test('is set to 1 by clear all user data action', () => {
+        test('is set to 0 by clear all user data action', () => {
             const oldStore = new UserProfileBuilder().build();
             const newStore = reducer(oldStore, clearAllUserData());
-            expect(newStore.timesAppStarted).toBe(1);
+            expect(newStore.timesAppStarted).toBe(0);
```
Comment (rasmus-storjohann-PG) [inline src/stores/__tests__/user_profiles.test.ts:223]: that makes more sense

File: src/components/main/main_component.tsx [hunk 6]
```diff
@@ -30,6 +30,7 @@ import { RegionDrawerConnectedComponent } from '../header_menu/region_drawer_con
 import { BackHandler, SafeAreaView } from 'react-native';
 import { UxTheme } from '../../application/ux_themes';
 import { ApplicationStartedAction } from '../../stores/user_profile';
+import { hasAction, requestReview } from 'expo-store-review';
```
Comment (rasmus-storjohann-PG) [inline src/components/main/main_component.tsx:33]: Odd names for these functions, you can make better names at the point of import : `import { foo as bar, bz as fuz } from 'whatever';`

File: src/components/main/main_component.tsx [hunk 7]
```diff
@@ -116,6 +120,17 @@ export const MainComponent = (props: Props): JSX.Element => {
         return (): void => subscription.remove();
     }, []);
 
+    useEffect((): void => {
+        sendReviewRequest();
+        props.applicationStarted();
+    }, []);
```
Comment (rasmus-storjohann-PG) [inline src/components/main/main_component.tsx:126]: Do you think this is consistent with the single responsibility principle? Let's talk

**Review by rasmus-storjohann-PG — CHANGES_REQUESTED** [review]
Very nice work

File: src/components/main/main_connected_component.ts (hunk 4 above)
Comment (brandon-PG) [reply src/components/main/main_connected_component.ts:59]: selectTimesAppStarted in needed in `select_user_data_for_local_persistence.ts` when saving the value to persistent data. I either create another selector or keep this logic in the connected component.

**Pull request merged**